### PR TITLE
AzureMonitor: Display variables in resource picker for Azure Monitor Logs

### LINKED ISSUE #34590: AzureMonitor: List template variables in Resource Picker

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

Lists template variables in the Azure Monitor Logs resource picker

image
image

**Which issue(s) this PR fixes**:

Fixes #34590

### CODE REVIEW

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourcePicker/index.tsx [hunk 1]
```diff
@@ -5,29 +5,40 @@ import { css } from '@emotion/css';
...
   const [internalSelected, setInternalSelected] = useState<string | undefined>(resourceURI);
 
   // Sync the resourceURI prop to internal state
   useEffect(() => {
     setInternalSelected(resourceURI);
   }, [resourceURI]);
 
+  const rows = useMemo(() => {
+    const templateVariableRow = resourcePickerData.transformVariablesToRow(templateVariables);
+    return [...azureRows, templateVariableRow];
```
Comment (sunker) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourcePicker/index.tsx:39]: What if there are no variables? I guess we don't want to add a row for template variables then.

**Review by sarahzinger — APPROVED** [review]
lgtm!

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourcePicker/utils.ts [hunk 2]
```diff
@@ -34,3 +35,16 @@ export function findRow(rows: ResourceRowGroup, id: string): ResourceRow | undef
...
   return undefined;
 }
+
+export function addResources(rows: ResourceRowGroup, targetResourceGroupID: string, newResources: ResourceRowGroup) {
+  return produce(rows, (draftState) => {
+    const draftRow = findRow(draftState, targetResourceGroupID);
+
+    if (!draftRow) {
+      // This case shouldn't happen often because we're usually coming here from a resource we already have
+      throw new Error('Unable to find resource');
```
Comment (sunker) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourcePicker/utils.ts:45]: This actually happened when no template variables were defined.

**Review by sunker — APPROVED** [review]
Fantastic! Just need to handle that case when there's no variables defined. Either remove the template variable group in that case, or handle error in NestedRows.onRowToggleCollapse

**Pull request merged**